lib/compat/wordpress-6.2/get-global-styles-and-settings.php

```diff
@@ -9,39 +9,66 @@
 	/**
 	 * Whether a theme or its parent have a theme.json file.
 	 *
-	 * @param boolean $clear_cache Whether the cache should be cleared and theme support recomputed. Default is false.
+	 * The result would be cached via the WP_Object_Cache.
+	 * It can be cleared by calling wp_theme_has_theme_json_clean_cache().
 	 *
 	 * @return boolean
 	 */
-	function wp_theme_has_theme_json( $clear_cache = false ) {
-		static $theme_has_support = null;
+	function wp_theme_has_theme_json() {
```

Comment:
This function should accept a theme as a param. So we can check if none active themes have theme json. Example. 
https://github.com/WordPress/wordpress-develop/blob/551c3d6619b7e88742911ca2dd3aed77a71cb0dc/src/wp-includes/rest-api/endpoints/class-wp-rest-themes-controller.php#L368-L374

Reply:
That could be useful, although I'm reluctant to do it in this PR. All the current use cases we have don't need it, so it adds complexity (manage invalidation of stylesheet per theme) without real needs. When/If we need that, we should implement it. It's a good follow-up PR.

Reply:
I agree with both of you here, but for now I agree more with @oandregal that this can be added in a follow up PR.

Reply:
[Here](https://github.com/WordPress/wordpress-develop/blob/4e72f78c75819470cb8f6d5d5d16f9ef8c9a7a0c/src/wp-includes/class-wp-theme-json-resolver.php#L423-L425) and [here](https://github.com/WordPress/wordpress-develop/blob/4e72f78c75819470cb8f6d5d5d16f9ef8c9a7a0c/src/wp-includes/rest-api/endpoints/class-wp-rest-themes-controller.php#L372) are examples where we need to know if another theme that is not the current theme has theme supports json. 

We can add this later, but this is something that is required.